### DISCUSSION TITLE
Conversion between grams and tonnes

### DISCUSSION THREAD

**Discussion by bennidhamma** [josdejong/mathjs#3179] [Q&A]
Is there any easy way to convert / format between grams and tonnes? I am doing some formatting and getting 'Mg' instead of 't' as a prefix, and having to go through some contortions to get the grams to convert into tonnes (and then once I do that, subsequent formats are also odd, like something in kg is then formatted as being mt, which is not what I want...)

**Comment by josdejong** [comment]
You can use `to` to convert to another unit:

```js
math.unit('907.18474 Mg').to('ton').toString() // "1000 ton"

math.evaluate('907.18474 Mg to ton').toString() // "1000 ton"
```

**Reply by bennidhamma** [reply]
Thank you @josdejong. (and thanks for this awesome project, I'm having so much fun with it)

i'm sorry I wasn't more clear on my use-case. The issue is that I don't necessarily know the unit that's passing through the function. It might be grams, but it might also be liters (working on some environmental factors). So I'm letting the user pick the units from a UI to interact with. then, math happens, and their numbers get big, and suddenly it's in the 1000s of kilograms, and when I format that value, basically like:

```
function printUnit(u: Unit) {
  print('<div>' + format(u) + '</div>');
}
```

I end up with Mg. what's curious, is that if I do something fancier, like let the user pick a 'big unit' in the UI (what I'm currently doing), and then checking to see if u is bigger than that unit, kind of like this:

```
function printUnit(u: Unit, bigUnit: Unit /* bigUnit is valueless, like unit('tonne') */ ) {
  if (evaluate(`${format(u)} > 1 ${format(bigUnit)}`)) {
    print(format(evaluate(`${format(u) to ${format(bigUnit)`});
  } else {
     print('<div>' + format(u) + '</div>');
  }
}
```

this kind works, but what oddly seems to happen here is that mathjs then reports smaller values in terms of 'mT' instead of kg...

Does that make sense at all?

**Reply by josdejong** [reply]
Can you give a concrete example of where mathjs reports a result in `mT` where you would expect `kg`?
